tests/functional/scrapyd_run.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python
+
+# -*- coding: utf-8 -*-
+#
+# This file is part of hepcrawl.
+# Copyright (C) 2015, 2016, 2017 CERN.
+#
+# hepcrawl is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+
+
+"""This module mocks the way that scrapyd runs the txapp in order to enable coverage"""
+
+
+from twisted.scripts.twistd import run
+from os.path import join, dirname, realpath
+from sys import argv
+
+import atexit
+import coverage
+
+cov = coverage.Coverage()
+cov.start()
+
+
+def save_coverage():
+    cov.stop()
+    cov.save()
+
+atexit.register(save_coverage)
+
+
+if __name__ == "__main__":
+    argv[1:1] = ['-n', '-y', join(dirname(realpath(__file__)), 'txapp.py')]
```

Comment:
Instead of `__file__` and having the txapp.py extra file, you can use `scrapyd.__file__`.